contracts/IdentityOracle.sol

```diff
@@ -0,0 +1,308 @@
+    // This function is called only oracle
+    //- only approved oracles can set the new merkle state plus link to ipfs data used to create state
+    {
+        _onlyOracle();
+        tmpStateHash = _statehash;
+        Chainlink.Request memory req = buildChainlinkRequest(
+            stringToBytes32(JOBID_GET_IPFS_CID1),
+            address(this),
+            this.setTmpStateDataIPFS1.selector
+        );
+        sendChainlinkRequestTo(ORACLE_ADDRESS, req, ORACLE_PAYMENT);
+    }
+
+    // It's the third function to be called by the oracle to store the first 32 bytes of IPFS CID
+    function setTmpStateDataIPFS1(bytes32 _requestId, bytes32 _ipfscid1)
+        public
+        recordChainlinkFulfillment(_requestId)
+    // This function is called only oracle
+    //- only approved oracles can set the new merkle state plus link to ipfs data used to create state
+    {
+        _onlyOracle();
+        tmpStateDataIPFS1 = _ipfscid1;
+        Chainlink.Request memory req = buildChainlinkRequest(
+            stringToBytes32(JOBID_GET_IPFS_CID2),
+            address(this),
+            this.completeIPFSandStateHashProcess.selector
+        );
+        sendChainlinkRequestTo(ORACLE_ADDRESS, req, ORACLE_PAYMENT);
+    }
+    
+    // It's the fourth function to be called by the oracle to store the second 32 bytes of IPFS CID
+    // And then assign the StateHash and IPFS CID values at once
+    function completeIPFSandStateHashProcess(
+        bytes32 _requestId,
+        bytes32 _ipfscid2
+    )
+        public
+        recordChainlinkFulfillment(_requestId)
+    // This function is called only oracle
+    //- only approved oracles can set the new merkle state plus link to ipfs data used to create state
+    {
+        _onlyOracle();
+        tmpStateDataIPFS2 = _ipfscid2;
+        setState(
+            tmpStateHash,
+            convIPFSCIDinBytes32ToStr(tmpStateDataIPFS1, tmpStateDataIPFS2)
+        );
+    }
+
+    //- prove that pair publicAddress, lastAuthenticated exists in current state.
+    //update address state in smart contract. also update address lastProofDate (required by isWhitelisted below).
+    //Proof can be generated by "sdk" defined in previous step.
+    function prove(
+        address _address,
+        uint256 _lastAuthenticated,
+        bytes32[] memory _proof
+    ) public {
+        bool result = false;
+        if (isValidProof(_address, _lastAuthenticated, _proof)) {
+            result = true;
+        } else {
+            ( , result) = _checkMerkleProof(
+                _address,
+                _lastAuthenticated,
+                stateHash,
+                _proof
+            );
+            //update address state in smart contract. also update address lastProofDate (required by isWhitelisted below).
+            if (result) {
+                WhitelistProofState memory state;
+                state.proof = _proof;
+                state.lastProofDate = block.timestamp;
```

Comment:
not required, definitely dont store the _proof
just store the address state as described in the specifications

Reply:
fixed with https://github.com/GoodDollar/ChainlinkIdentityOracle/pull/4/commits/c3975466278f5b259ccd825b6ba137a0404a1845